internal/manifest/iso_tree.go

```diff
@@ -148,6 +135,17 @@ func bootISOMonoStageOptions(kernelVer, arch, vendor, product, osVersion, isolab
 	}
 }
 
+type ISOTreePayload interface {
```

Comment:
Not sure I love this abstraction. Really what we want is a proper union, but golang... How about we just make both the `OSTree` and the `Payload` arguments (which should be of type `pipelien.OS`) optional, assert on exactly one being set, and then just generate the right stages in the pipeline function?